benchmarks/scripts/benchmark_field_last.jl

```diff
@@ -0,0 +1,333 @@
+│     FLD.aos_cart_offset!(X_aos_ref, Y_aos_ref, us; bm, nreps = 100) │ 61 microseconds, 185 nanoseconds │ 32.5079 │ 662.837     │ 2              │ 100    │
+│     FLD.aos_lin_offset!(X_aos, Y_aos, us; bm, nreps = 100)          │ 31 microseconds, 376 nanoseconds │ 63.3926 │ 1292.57     │ 2              │ 100    │
+│     FLD.soa_linear_index!(X_soa, Y_soa, us; bm, nreps = 100)        │ 31 microseconds, 120 nanoseconds │ 63.9141 │ 1303.21     │ 2              │ 100    │
+│     FLD.soa_cart_index!(X_soa, Y_soa, us; bm, nreps = 100)          │ 44 microseconds, 53 nanoseconds  │ 45.1499 │ 920.607     │ 2              │ 100    │
+└─────────────────────────────────────────────────────────────────────┴──────────────────────────────────┴─────────┴─────────────┴────────────────┴────────┘
+```
+
+# CPU (Mac M1)
+```
+[ Info: ArrayType = identity
+Problem size: (63, 4, 4, 5400, 1), float_type = Float32, device_bandwidth_GBs=2039
+┌─────────────────────────────────────────────────────────────────────┬───────────────────────────────────┬──────────┬─────────────┬────────────────┬────────┐
+│ funcs                                                               │ time per call (CPU)               │ bw %     │ achieved bw │ n-reads/writes │ n-reps │
+├─────────────────────────────────────────────────────────────────────┼───────────────────────────────────┼──────────┼─────────────┼────────────────┼────────┤
+│     FLD.aos_cart_offset!(X_aos_ref, Y_aos_ref, us; bm, nreps = 100) │ 16 milliseconds, 494 microseconds │ 0.241171 │ 4.91747     │ 4              │ 100    │
+│     FLD.aos_lin_offset!(X_aos, Y_aos, us; bm, nreps = 100)          │ 783 microseconds, 256 nanoseconds │ 5.07871  │ 103.555     │ 4              │ 100    │
+│     FLD.soa_linear_index!(X_soa, Y_soa, us; bm, nreps = 100)        │ 790 microseconds, 894 nanoseconds │ 5.02966  │ 102.555     │ 4              │ 100    │
+│     FLD.soa_cart_index!(X_soa, Y_soa, us; bm, nreps = 100)          │ 12 milliseconds, 522 microseconds │ 0.317663 │ 6.47714     │ 4              │ 100    │
+└─────────────────────────────────────────────────────────────────────┴───────────────────────────────────┴──────────┴─────────────┴────────────────┴────────┘
+```
+
+=#
+
+#! format: off
+module BenchmarkFieldLastIndex
+
+include("benchmark_utils.jl")
+
+@inline function const_linear_index(us::UniversalSizesStatic, I, field_index)
+    n = (get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), 1)
+    i = I + prod(n)*field_index
+    return i
+end
+
+@inline function const_linear_index_reference(us::UniversalSizesStatic, I, field_index)
+    CI = CartesianIndices((get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), 1))
+    LI = LinearIndices((get_Nv(us), get_Nij(us), get_Nij(us), get_Nh(us), field_index+1))
+    return LI[CI[I] + CartesianIndex((0, 0, 0, 0, field_index))]
+end
+
+# add3(x1, x2, x3) = x1 + x2 + x3
```

Comment:
Should this be uncommented?

Reply:
Yes, I did not loop over all benchmarks.

This corresponds to the `n_reads_writes = 2` benchmark.

Reply:
Sorry, I updated my response, @sriharshakandala.